.github/workflows/sync-end-to-end.yml

```
@@ -43,62 +43,74 @@ jobs:
           -destination "platform=iOS Simulator,name=iPhone 14" \
           -derivedDataPath "DerivedData" \
         | tee xcodebuild.log
+
+    - name: Store Binary
+      uses: actions/upload-artifact@v3
+      with:
+        name: duckduckgo-ios-app
+        path: DerivedData/Build/Products/Debug-iphonesimulator/DuckDuckGo.app
     
+    - name: Upload logs when workflow failed
+      uses: actions/upload-artifact@v3
+      if: failure()
+      with:
+        name: BuildLogs
+        path: |
+          xcodebuild.log
+          DerivedData/Logs/Test/*.xcresult
+        retention-days: 7
+    
+  sync-end-to-end-tests:
+    name: Sync End To End Tests
+    needs: build-for-sync-end-to-end-tests
+    runs-on: macos-13
+    strategy:
```

There's one more argument that we can pass to `strategy` which is `fail-fast: false`. By default it's `true` and whenever any of the concurrent jobs fails, all others that are still in progress are cancelled. When we set it to `false`, all concurrent jobs will run to finish (or error) without looking at other jobs. I think it would be useful to check whether a failure happens on a specific iOS version. What do you think @loremattei?

```suggestion
    strategy:
      fail-fast: false
```